### ISSUE TITLE
Unable to push with android client

### ISSUE THREAD

**Issue by esell** [laurent22/joplin#249]
### Operating system

- [ ] Windows
- [ ] macOS
- [ ] Linux
- [x] Android
- [ ] iOS

### Application

- [ ] Desktop
- [x] Mobile
- [ ] Terminal

I am running into an odd issue where syncing from the android mobile client isn't working 100%. The steps are:

* Configure desktop (macOS) client, push empty notebook to webdav server (works)
* Configure android client, sync with webdav server to get empty notebook via webdav (works)
* Add note on android client, attempt to sync with webdav server, 403 error occurs (fails)

The odd part to me is that everything works 100% on the macOS desktop client, it is only the android client that is having issues. I uninstalled and re-installed the android client just to make sure but still see the same results. Is it possible that there is a specific config on the server side that needs to be done? If so, how come the desktop client works? The debug log from the android client is attached.
[syncReport-1519172741103.txt](https://github.com/laurent22/joplin/files/1742424/syncReport-1519172741103.txt)

**Comment by laurent22** [comment]
Which WebDAV server is it? And if you click on "Check configuration" from the Config screen is it showing any error?

Is it http or https? If it's https, is it using a self-signed certificate?

All the apps share the same sync code so indeed if one works in theory another should too, but in practice the network stack is different from one device to another, and that might need to worked around (for instance, iOS sends additional headers that were breaking certain WebDAV servers).

**Comment by esell** [comment]
I am using the webdav portion of seafile. When I test the config it reports that everything is working. It is over https but I am using a cert from let's encrypt  (not self signed). Let me know if there is anything you would like me to test/look at.

**Comment by laurent22** [comment]
Saefile seems to work for other users so it could be something specific to the server. Is there any test account you could share so that I can try to replicate this? https://github.com/laurent22/joplin/blob/master/Assets/Adresse.png?raw=true

**Comment by fmrtn** [comment]
I use Seafile webdav as backend and works fine with iOS and Android over https with let's encrypt certificate.

@esell you can get more info about your issue in your server logs, look at these files

seafdav.access.log
seafdav.error.log

the location depends on your configuration.

**Comment by esell** [comment]
thanks for confirming that it is probably on my end @fmrtn, i'll take a look and see if i can figure out what is going on.

@laurent22, thanks for the offer but i've already taken up enough of your time :). if i just can't figure it out i'll ping you but for now you probably have enough on your plate. i'll close this since it appears to be an issue outside of joplin. thanks for providing an awesome app.

**Comment by esell** [comment]
So I finally got some time to look at this and it appears that maybe the android app is sending the wrong password? Here is what the auth header looks like when i use the desktop client:

`ZXNlbGxAZXNoZWF2eWluZHVzdHJpZXMuY29tOk15UGFTcw0K` which decodes to:
`esell@esheavyindustries.com:MyPaSs`


However, when I use the android client the auth header looks like:

`ZXNlbGxAZXNoZWF2eWluZHVzdHJpZXMuY29tIDpNeVBhU3MNCg==` which decods to:
`esell@esheavyindustries.com :MyPaSs`


If you look closely you'll notice there is whitespace on the end of the username in the 2nd example. I went in and removed all of my credentials on the android app and entered them again but nothing changed. Is there a cache or something that Joplin might use on android that I might have to clear? Is this likely to be causing the issue?
